x-pack/plugins/security_solution/public/onboarding/components/onboarding_body/cards/integrations/package_list_grid.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Suspense, useCallback, useEffect, useRef, useState } from 'react';
+
+import { EuiButtonGroup, EuiFlexGroup, EuiFlexItem, EuiSkeletonText } from '@elastic/eui';
+import type { AvailablePackagesHookType, IntegrationCardItem } from '@kbn/fleet-plugin/public';
+import { noop } from 'lodash';
+
+import { css } from '@emotion/react';
+import { PackageList } from './utils';
+import { useIntegrationCardList, useTabMetaData } from './hooks';
+import {
+  useStoredIntegrationSearchTerm,
+  useStoredIntegrationTabId,
+} from '../../../../hooks/use_stored_state';
+import { useOnboardingContext } from '../../../onboarding_context';
+import {
+  DEFAULT_TAB,
+  INTEGRATION_TABS,
+  LOADING_SKELETON_HEIGHT,
+  SCROLL_ELEMENT_ID,
+  SEARCH_FILTER_CATEGORIES,
+  WITHOUT_SEARCH_BOX_HEIGHT,
+  WITH_SEARCH_BOX_HEIGHT,
+} from './const';
+
+interface WrapperProps {
+  useAvailablePackages: AvailablePackagesHookType;
+}
+
+const emptyStateStyles = { paddingTop: '16px' };
+export const PackageListGrid = React.memo(({ useAvailablePackages }: WrapperProps) => {
+  const { spaceId } = useOnboardingContext();
+  const scrollElement = useRef<HTMLDivElement>(null);
+  const [selectedTabId, setSelectedTabIdToStorage] = useStoredIntegrationTabId(
+    spaceId,
+    DEFAULT_TAB.id
+  );
+  const [searchTermFromStorage, setSearchTermToStorage] = useStoredIntegrationSearchTerm(spaceId);
```

Comment:
A user may store the search term in a specific tab and when the page is reloaded be used in a different tab, which may no longer make sense. Is this a requirement?

Reply:
I currently have the same search term persisted across tabs, it is only cleared on the recommended tab when the search box is hidden to avoid filtered results. (The search term is recovered after visiting the recommended tab as well) In this case, when users coming from the Fleet page, users would have the same results in the tab content. What do you think?


https://github.com/user-attachments/assets/19619a71-19ca-421d-9f1b-bcc6d6944c6e

Reply:
Okay, that sounds great, thanks for the explanation